controllers/users.js

```diff
@@ -0,0 +1,53 @@
+const auth = require('../auth');
+
+exports.register = async (req, res) => {
+  const { username, password } = req.body;
+  if (username && password) {
```

Comment:
Following the idea of Controllers just telling people what to do, we should follow the **_Tell, Don't Ask_** principle.

It just means that you ask whatever thing is responsible of doing it to do it, and it's the other object's responsibility to validate it:
```js
exports.register = async (req, res) => {
  const { username, password } = req.body;
  
  try {
    const user = await RegisterUser.call({ username, password });
    res.status(201).json({ user });
  catch (error) {
    res.status(422).json({ error });
  }
};